auth/store/mongo/provider_session_repository.go

```diff
@@ -127,28 +135,30 @@ func (p *ProviderSessionSession) DeleteAllProviderSessions(ctx context.Context,
 	return nil
 }
 
-func (p *ProviderSessionSession) GetProviderSession(ctx context.Context, id string) (*auth.ProviderSession, error) {
+func (p *ProviderSessionRepository) GetProviderSession(ctx context.Context, id string) (*auth.ProviderSession, error) {
 	if ctx == nil {
 		return nil, errors.New("context is missing")
 	}
 	if id == "" {
 		return nil, errors.New("id is missing")
 	}
 
-	if p.IsClosed() {
-		return nil, errors.New("session closed")
-	}
-
 	now := time.Now()
 	logger := log.LoggerFromContext(ctx).WithField("id", id)
 
 	providerSessions := auth.ProviderSessions{}
-	err := p.C().Find(bson.M{"id": id}).Limit(2).All(&providerSessions)
+	opts := options.Find().
+		SetLimit(2)
```

Comment:
There's a unique index on the id field - this query cannot return more than 1 record.

Reply:
Good catch! That's pre-existing, but we should definitely fix it while we're here. 👍